vm/src/bytesinner.rs

```diff
@@ -410,6 +416,48 @@ impl PyBytesInner {
         self.elements.delete_slice(vm, &slice)
     }
 
+    pub fn pop(&mut self, index: isize, vm: &VirtualMachine) -> PyResult<u8> {
+        if let Some(index) = self.elements.get_pos(index) {
+            Ok(self.elements.remove(index))
+        } else {
+            Err(vm.new_index_error("index out of range".to_owned()))
+        }
+    }
+
+    pub fn insert(
+        &mut self,
+        index: isize,
+        object: PyObjectRef,
+        vm: &VirtualMachine,
+    ) -> PyResult<()> {
+        let value = Self::value_try_from_object(vm, object)?;
+        let index = get_bounded_pos(index, self.len());
+        self.elements.insert(index, value);
+        Ok(())
+    }
+
+    pub fn append(&mut self, object: PyObjectRef, vm: &VirtualMachine) -> PyResult<()> {
+        let value = Self::value_try_from_object(vm, object)?;
+        self.elements.push(value);
+        Ok(())
+    }
+
+    pub fn remove(&mut self, object: PyObjectRef, vm: &VirtualMachine) -> PyResult<()> {
+        let value = Self::value_try_from_object(vm, object)?;
+        if let Some(index) = self.elements.iter().position(|&x| x == value) {
```

Comment:
`self.elements.find(value)` may work

Reply:
Vec.find return a reference to the item and we can't do deleting.

Reply:
Will you check bstr's find? It returns index instead of reference.
I think we use it to implement bytes.find too.

https://docs.rs/bstr/0.2.13/bstr/trait.ByteSlice.html#method.find

Reply:
Thanks for the suggestion, I replaced it by bstr function.